### ISSUE TITLE
ICP registration buttons throw error in gui/draw.py example

### ISSUE THREAD

**Issue by ShreyanshDarshan** [isl-org/Open3D#4531]
## Bug Description:
On clicking `ICP Registration (one set)` button in the `Custom Actions` section, the terminal throws the error:
```
Traceback (most recent call last):
  File "draw.py", line 344, in <module>
    main()
  File "draw.py", line 340, in main
    selections()
  File "draw.py", line 198, in selections
    vis.draw([{
  File "/home/darshan/miniconda3/envs/open3d/lib/python3.8/site-packages/open3d/visualization/draw.py", line 123, in draw
    gui.Application.instance.run()
  File "draw.py", line 168, in do_icp_one_set
    source_picked = sorted(list(sets[0][source_name]),
IndexError: list index out of range
```

On clicking `ICP Registration (two sets)` button in the `Custom Actions` section, the terminal throws the error:
```
Traceback (most recent call last):
  File "draw.py", line 344, in <module>
    main()
  File "draw.py", line 340, in main
    selections()
  File "draw.py", line 198, in selections
    vis.draw([{
  File "/home/darshan/miniconda3/envs/open3d/lib/python3.8/site-packages/open3d/visualization/draw.py", line 123, in draw
    gui.Application.instance.run()
  File "draw.py", line 184, in do_icp_two_sets
    source_set = sets[0][source_name]
IndexError: list index out of range
```

## Steps to reproduce the behavior:

```sh
cd  Open3D/examples/python/gui
python draw.py
```
A visualizer window will pop up.
Close the visualizer window 3 times to get to the icp registration example.
Then click on the button called `ICP Registration (one set)` or `ICP Registration (two sets)` under the `Custom Actions` section.

## Expected behaviour:
The yellow and blue pointclouds were expected to align together.

## Screenshots:
![image](https://user-images.githubusercontent.com/41020033/148028407-3d99b5b3-2e3e-449f-b8c5-9ab87390e319.png)

## Environment:

- Operating system: Ubuntu 20.04
- Python version: Python 3.8.12
- Open3D version: 0.14.1
- Is this remote workstation?: no
- How did you install Open3D?: build from source
- Compiler version (if built from source): gcc 9.3.0

**Comment by reyanshsolis** [comment]
There seems to be some issue with selecting the points. Will confirm the same with Rene.

**Comment by errissa** [comment]
@ShreyanshDarshan Can you test this again with 0.15? A PR during the last cycle fixed an issue with picking.

**Comment by ShreyanshDarshan** [comment]
@errissa you are right, it is fixed now in 0.15. Thanks!
